src/core/endpoints/subscribe.js

```diff
@@ -30,7 +30,7 @@ export function isAuthSupported() {
 }
 
 export function prepareParams({ config }: ModulesInject, incomingParams: SubscribeArguments): Object {
-  let { channelGroups = [], timetoken, filterExpression, region } = incomingParams;
+  let { state, channelGroups = [], timetoken, filterExpression, region } = incomingParams;
```

Comment:
It's probably safer to set `state = {}` in case it was not passed into the `incomingParams` this way later you won't get an error when you do `Object.keys(state)`

Reply:
@davidnub it is good and not so good idea to change. 
This value is passed from internal SDK logic. Subscriber provides placeholder (`{}`) if there is no channels with state.
If because of some reasons `Object.kets(state)` will throw, because of attempt to access `undefined` - it will mean, what something has been messed up. If we will add default value in `destructure` - we can miss this potentially bug-prone change.

Reply:
Yes I agree with you that the value passed in from the internal SDK logic will include an empty object and therefore we _shouldn't_ need to worry about it. This is more of a "best practice" scenario to treat each function as its own unit and therefore securing it will only prove that this specific function operates as expected.

Detecting an error _only_ when the calling function changes smells more like "testing in real time" which is not great, but I'm fine to let go to ensure we have progress. Specifically since other destructured properties are not being defaulted either.